layers/+lang/c-c++/packages.el

```diff
@@ -145,3 +135,9 @@
     :post-init
     (dolist (mode '(c-mode c++-mode))
       (spacemacs/setup-helm-cscope mode))))
+
+(defun c-c++/post-init-gdb-mi ()
+  (setq debug-mode 'gdb)
+  (if c-c++-debugger-prog
+      (setq debug-prog c-c++-debugger-prog)
+    (setq debug-prog "gdb -i=mi")))
```

Comment:
`(setq debug-prog (or c-c++-debugger-prog "gdb -i=mi"))`

Or better yet, make the default value for `c-c++-debugger-prog` equal ot `"gbd -i=mi"` so you don't have to do this.